src/Stratis.Bitcoin/P2P/PeerDiscoveryLoop.cs

```diff
@@ -96,7 +96,15 @@ public void DiscoverPeers(IConnectionManager connectionManager)
 
             this.currentParameters = connectionManager.Parameters.Clone(); // TODO we shouldn't add all the behaviors, only those that we need.
 
-            this.asyncLoop = this.asyncProvider.CreateAndRunAsyncLoop(nameof(this.DiscoverPeersAsync), async token =>
+            this.discoverFromDnsSeedsLoop = this.asyncProvider.CreateAndRunAsyncLoop(nameof(this.DiscoverFromDnsSeedsAsync), async token =>
+            {
+                if (this.peerAddressManager.Peers.Count < TargetAmountOfPeersToDiscover)
+                    await this.DiscoverFromDnsSeedsAsync();
+            },
+            this.nodeLifetime.ApplicationStopping,
+            TimeSpan.FromHours(1));
```

Comment:
So basically every node on then network will hit the dns seedrs every hour if target peer not reached? Currently stratis has no more then abut 300 nodes on the network.

Reply:
No, DNS discovery is only performed if necessary, i.e. if there are no known peers, or if all the known peers have been attempted (i.e. all of them have failed to be connected to).